packages/scripts/scripts/test-e2e.js

```diff
@@ -0,0 +1,45 @@
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.BABEL_ENV = 'test';
+process.env.NODE_ENV = 'test';
+
+// Makes the script crash on unhandled rejections instead of silently
+// ignoring them. In the future, promise rejections that are not handled will
+// terminate the Node.js process with a non-zero exit code.
+process.on( 'unhandledRejection', ( err ) => {
+	throw err;
+} );
+
+/**
+ * External dependencies
+ */
+const jest = require( 'jest' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	fromConfigRoot,
+	getCliArgs,
+	hasCliArg,
+	hasProjectFile,
+	hasJestConfig,
+} = require( '../utils' );
+
+// Provides a default config path for Puppeteer when jest-puppeteer.config.js
+// wasn't found at the root of the project or a custom path wasn't defined
+// using JEST_PUPPETEER_CONFIG environment variable.
+if ( ! hasProjectFile( 'jest-puppeteer.config.js' ) && ! process.env.JEST_PUPPETEER_CONFIG ) {
+	process.env.JEST_PUPPETEER_CONFIG = fromConfigRoot( 'puppeteer.config.js' );
+}
+
+const config = ! hasJestConfig() ?
```

Comment:
For end-to-end tests, should this be checking for `jest.config.js` (or `package.json` `jest`)? Or only `jest-puppeteer.config.js`? How should we expect to give people control over whether to override configuration for either unit tests or end-to-end tests?

I'm running into this in my [Dones project](https://github.com/aduth/dones), where I do choose to [override the default configuration](https://github.com/aduth/dones/blob/25a4057de4736ac882b1eb0201431187d267f668/package.json#L90-L95) for unit tests, but unfortunately it seems to bleed over into end-to-end test scripts as well, where I _do_ simply want to use the default. Or at least, it's not clear how I can configure for both unit tests and end-to-end tests.

Reply:
> It kinda feels like this should be limited to checking for the presence of jest-e2e.config.js perhaps?

An interesting challenge and I'm surprised it didn't come up earlier :)

How about we add higher priority for non-standard:
- `jest-e2e.config.js`
- `jest-unit.config.js`

To give folks a way to target only one or both of those tools without the need for using `--config` flag. As a temporary workaround, you can use the non-standard name of the file and pass it as a config option to the test script.

Reply:
In retrospect, it seems a bit more obvious in retrospect in in observing that the [default configurations folder](https://github.com/WordPress/gutenberg/tree/master/packages/scripts/config) includes variations of these two files (`jest-e2e.config.js`, `jest-unit.config.js`), yet only one is supported for end-users. It's an interesting exercise to consider for other potential issues. Is there a similar problem with `.eslintrc-md.js` and `.eslintrc.js` ?

>How about we add higher priority for non-standard:

I think this could be a good solution. I was worried if it might have to be a "Breaking Change" if we only start considering one of these two, but (a) IIRC `jest.config.js` is a "standard" file supported by Jest and (b) it could be viewed as a rare, advanced use-case where one opts to configure both.

>As a temporary workaround, you can use the non-standard name of the file and pass it as a config option to the test script. In fact, this is how it works in Gutenberg:

Yeah, I ended up using something like this, or at least observing that higher priority is given to `--config`, also observing that the E2E configuration is quite small and I wasn't entirely on board with the `specs` forced folder convention of the default configuration 😅 

```
wp-scripts test-e2e --config {\"preset\":\"jest-puppeteer\"} e2e
```

Reply:
Pull request: #22477